target/linux/mediatek/image/mt7623.mk

```diff
@@ -1,5 +1,9 @@
 KERNEL_LOADADDR := 0x80008000
 
+FEATURES += ext4 usb
+DEVICE_PACKAGES := kmod-fs-vfat kmod-nls-cp437 kmod-nls-iso8859-1 kmod-mmc \
+	mkf2fs e2fsprogs kmod-usb-ohci kmod-usb2 kmod-usb3
```

Comment:
Ah, GitHub messed up the patch order again.

Anyway, DEVICE_PACKAGES should be defined per device. It won't work properly otherwise.
If you want to provide default packages for one subtarget, you have different options:
1. use the Device/Default definition in image/Makefile, which will be used as base setup for all devices (there already is an ifeq for a subtarget there)
2. create a new definition that will be used for mt7623 devices only and then included for the devices * (however, that's not really helpful just for one variable)
3. use DEFAULT_PACKAGES in mediatek/Makefile or mt7623/target.mk
3. Just keep the DEVICE_PACKAGES with the devices. Particularly for usb, I'd personally go that way.

Reply:
> 2.  create a new definition that will be used for mt7623 devices only and then included for the devices (see below, however, that's not really helpful just for one variable)

But then the other variables from the ifeq($(SUBTARGET),mt7623) block in `image/Makefile` could live there too, right? And the same variables in the `else` block could go into the other `mt76xx.mk` files close to the actual devices?

Reply:
One could go that road, yes, however note that you would need to have the "include" $(Device/mtsomething) for every devices. With the ifeq in in Device/Default, this is done automagically ...

I'd address from a conceptual point of view: Is it really a "default" for all/most devices? Then you should go with option 1 and/or 3. Is it a pattern that will be relevant for a class/group of devices, but not the entire subtarget (as far as you can predict)? Then go with 2. Is it actually device-specific, but happens to be the same for all relevant devices by coincidence? Then I'd tend to use option 4.
Of course, one may combine these options e.g. for different variables or packages ...

I'm aware that this is quite hard to decide if there is only two devices in the subtarget ...

Personally, I'm not a big fan of option 2, as this seems nice at the beginning, but will bite back at end (mostly because you add another layer between default device and device definition).

Reply:
Yeah, screw it. Someone is already adding a NAND-based mt7623 device anyway. I'll live with the duplication in `DEVICE_PACKAGES` for now. Thanks.